spec/services/book_services_spec.rb

```diff
@@ -86,15 +86,52 @@
     end
 
     describe '.add_book' do
-      xit 'adds a book to the shelf of a user' do
-        # how to test this functionality while mocking?
+      it 'adds a book to the shelf of a user' do
+        shelf_id = 4
+        key = Figaro.env.BOOK_KEY
+        volume_id = 'm8dPPgAACAAJ'
+
+        stub_request(:get, "https://books.googleapis.com/books/v1/mylibrary/bookshelves/#{shelf_id}/addVolume?key=#{key}&volumeId=#{volume_id}").
+          with(
+            headers: {
+        	  'Accept'=>'*/*',
+        	  'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+        	  'Authorization'=>'Bearer auth_token',
+        	  'User-Agent'=>'Faraday v1.5.1'
+            }).
+          to_return(status: 200, body: "{}", headers: {})
 
-        # volume_id = 'm8dPPgAACAAJ'
-        #
-        # VCR.turn_off!
-        #   actual = BookService.add_book(volume_id, 'auth_token')
-        # VCR.turn_on!
+        VCR.turn_off!
+          actual = BookService.add_book(shelf_id, volume_id, 'auth_token')
+        VCR.turn_on!
+
+        expect(actual.status).to eq(200)
+        expect(actual.body).to eq("{}")
```

Comment:
Thank you for adding these tests in! This makes sense to me - testing just that the response is successful with an empty body. I think I struggle fundamentally with stubbing out the service specs because it feels like we're not really testing anything  🤷‍♀️ .